snap/gadget_test.go

```diff
@@ -119,6 +120,36 @@ defaults:
       bar: baz
 `)
 
+var mockVolumeUpdateGadgetYaml = []byte(`
+volumes:
+  bootloader:
+    schema: mbr
+    bootloader: u-boot
+    id:     id,guid
+    structure:
+      - filesystem-label: system-boot
+        offset: 12345
+        offset-write: 777
+        size: 88888
+        type: id,guid
+        id:   id,guid
+        filesystem: vfat
+        content:
+          - source: subdir/
+            target: /
+            unpack: false
+          - image: foo.img
+            offset: 4321
+            offset-write: 8888
+            size: 88888
+            unpack: false
+        update:
+          edition: 5
+          preserve:
+           - env.txt
```

Comment:
Is this defined as case-sensitive or case-insensitive? Especially with FAT I think we should be careful about this.

Reply:
As discussed, we'll ignore the casing. I can check for duplicates, but that will probably come as a follow-up PR with actual validation of gadget.yaml.

Reply:
Should we make the case-sensitivity explicit in the definition?
```
update:
  edition: 5
  preserve-case-insensitive:
    - env.txt
```

Reply:
I think that `preserve` is fine as it is. Maybe something we could handle with documentation? Otherwise, I'd really want to avoid going full name bikeshedding cycle.